internal/handlers/pg/msg_findandmodify.go

```diff
@@ -117,49 +118,45 @@ func (h *Handler) MsgFindAndModify(ctx context.Context, msg *wire.OpMsg) (*wire.
 					return nil
 				}
 
+				var upsert *types.Document
+
 				if params.HasUpdateOperators {
 					upsert = resDocs[0].DeepCopy()
 					_, err = common.UpdateDocument(upsert, params.Update)
 					if err != nil {
 						return err
 					}
-
-					if _, err = updateDocument(ctx, tx, &queryParams, upsert); err != nil {
-						return err
-					}
 				} else {
 					upsert = params.Update
 
 					if !upsert.Has("_id") {
 						upsert.Set("_id", must.NotFail(resDocs[0].Get("_id")))
 					}
+				}
 
-					if _, err = updateDocument(ctx, tx, &queryParams, upsert); err != nil {
-						return err
-					}
+				if _, err = updateDocument(ctx, tx, &qp, upsert); err != nil {
+					return lazyerrors.Error(err)
 				}
-			}
 
-			var resultDoc *types.Document
-			if params.ReturnNewDocument || len(resDocs) == 0 {
```

Comment:
`len(resDocs) == 0` case returns early, see L109. So this is unreachable hence removing